components/Filters/RemoveFilter.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import {
+    Button,
+    ConfirmModal,
+    Alert,
+    useApiWithoutResult,
+    useModals,
+    useNotifications,
+    useEventManager
+} from 'react-components';
+import { deleteFilter } from 'proton-shared/lib/api/filters';
+import { noop } from 'proton-shared/lib/helpers/function';
+
+function RemoveFilter({ filter, className, onRemoveFilter }) {
+    const { call } = useEventManager();
+    const { createModal } = useModals();
+    const { createNotification } = useNotifications();
+    const { request, loading } = useApiWithoutResult(deleteFilter);
+
+    const handleConfirmConfirmModal = async () => {
+        await request(filter.ID);
+        await call();
+        createNotification({
+            text: c('Filter notification').t`Filter removed`
+        });
+        onRemoveFilter(filter);
+    };
+
+    const handleClick = () =>
+        createModal(
+            <ConfirmModal loading={loading} onConfirm={handleConfirmConfirmModal} title={c('Title').t`Delete Filter`}>
```

Comment:
This `loading` is useless because we close the modal when the user click on confirm. The loading state should be present on the action which open the modal.